### DISCUSSION TITLE
dpf using parallel computing

### DISCUSSION THREAD

**Discussion by Indrajeetrt** [ansys/pydpf-core#1828] [Q&A]
Hello there, Good day... I have a ansys dpf script which is taking more time than expected and i have done optimization of my code to it's best possible way but still the time taken to execute is high. I have found that we can start the local server as global server with the following code "dpf.start_local_server(as_global=True)" but i am unable to mention number of cores. Can you please help me on this ? and it would be good if you could share any examples on connecting it to HPC for parallel computing. Thank you in advance!!!

**Comment by sandeep-tubs** [comment]
Hi Indra, Again I am also needing this. I have a 30+ core performance machine. Would like to know if I can run on it using parallel computing. I hope someone answers how to do this.

**Comment by MeenaBytes** [comment]
@sandeep-tubs @Indrajeetrt ,

Please try this:

```
global_server = dpf.start_local_server(
    as_global=True, config=dpf.AvailableServerConfigs.InProcessServer
)

core_config = dpf.settings.get_runtime_core_config(server=global_server)
num_threads = core_config.num_threads
core_config.num_threads = 8
```

**Reply by xcast-ppi** [reply]
Hi @MeenaBytes @sandeep-tubs , did you notice any performance improvement when using multiple threads? I didn't and am wondering if something is not well setup here. Thanks
